platform/configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/FeatureOperation.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator.impl;
+
+import static org.apache.karaf.features.FeaturesService.Option.NoAutoRefreshBundles;
+
+import java.util.EnumSet;
+
+import org.apache.karaf.features.Feature;
+import org.apache.karaf.features.FeatureState;
+import org.apache.karaf.features.FeaturesService;
+import org.codice.ddf.admin.configurator.ConfiguratorException;
+import org.codice.ddf.admin.configurator.Operation;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Transactional handler for starting and stopping features.
+ */
+public class FeatureOperation implements OperationBase, Operation<Void, Boolean> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FeatureOperation.class);
+
+    private final String featureName;
+
+    private final boolean newState;
+
+    private final BundleContext bundleContext;
+
+    private final boolean initActivationState;
+
+    private FeatureOperation(String featureName, boolean newState, BundleContext bundleContext) {
+        this.featureName = featureName;
+        this.newState = newState;
+        this.bundleContext = bundleContext;
```

Comment:
Same comment here on bundle context, just want to make sure this is dynamic and not being injected via blueprint or only set once (even if you're using FrameworkUtil somewhere).